src/h5web/vis-packs/core/complex/hooks.ts

```diff
@@ -0,0 +1,74 @@
+import ndarray from 'ndarray';
+import { useMemo } from 'react';
+import type { DimensionMapping } from '../../../dimension-mapper/models';
+import type { H5WebComplex } from '../../../providers/models';
+import { useMappedArray } from '../hooks';
+import { Bounds, Domain, ScaleType } from '../models';
+import { getDomainValidForScale, getNewBounds } from '../utils';
+
+export function usePhaseAmplitude(
+  value: H5WebComplex[],
+  dims: number[],
+  dimMapping: DimensionMapping,
+  scaleType: ScaleType = ScaleType.Linear
+): {
+  phaseArray: ndarray;
+  phaseDomain: Domain | undefined;
+  amplitudeArray: ndarray;
+  amplitudeDomain: Domain | undefined;
+} {
+  const [mappedArray] = useMappedArray(value, dims, dimMapping);
```

Comment:
Seems unnecessary and inconsistent with the other mapped vis to have this call to `useMappedArray` here instead of inlined in the mapped vis, especially since it requires passing two more props to `usePhaseAmplitude`.